waspc/waspls/src/Wasp/LSP/Util.hs

```diff
@@ -54,3 +59,9 @@ getPathRelativeToProjectDir file = do
   case maybeProjectRootDir of
     Nothing -> pure Nothing
     Just projectRootDir -> pure $ stripProperPrefix projectRootDir file
+
+getWaspDirFromWaspFileUri :: LSP.Uri -> Maybe (SP.Path' SP.Abs (SP.Dir WaspProjectDir))
+getWaspDirFromWaspFileUri uri = lspUriToPath uri >>= Just . SP.parent
+
+lspUriToPath :: LSP.Uri -> Maybe (SP.Path' SP.Abs (SP.File ()))
+lspUriToPath = SP.parseAbsFile <=< stripPrefix "file://" . T.unpack . LSP.getUri
```

Comment:
Are you sure it will always be "file://", or is that just something you observed as a possible instance? Will it also be "file://" if vscode is running on Windows? Maybe it will, I don't know how vscode work with its uris, maybe it guarantees they are always Posix and follow this format, just wanted to make sure you covered that.

I know we are not currently supporting Windows, and also that this code might all go away relatively soon, but doesn't matter, we want to do it properly. So either we figure out how to do it to work in all cases, or we leave a NOTE explaining the situation.

Reply:
This is part of the previous implementation, I've just extracted it.

It seems from the LS specification, it will always be an URI with the `file://` prefix: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uri